doc/source/developing/testing.rst

```diff
@@ -148,358 +107,291 @@ functionality.
-    ``@requires_ds(test_dataset_name)``. This decorator can accept the
-    argument ``big_data=True`` if the test is expensive. The
-    ``test_dataset_name`` should be a string containing the path you would pass
-    to the ``yt.load`` function. It does not need to be the full path to the
-    dataset, since the path will be automatically prepended with the location of
-    the test data directory.  See :ref:`configuration-file` for more information
-    about the ``test_data-dir`` configuration option.
-
-  * There are ``small_patch_amr`` and ``big_patch_amr`` routines that you can
-    yield from to execute a bunch of standard tests. In addition we have created
-    ``sph_answer`` which is more suited for particle SPH datasets. This is where
-    you should start, and then yield additional tests that stress the outputs in
-    whatever ways are necessary to ensure functionality.
-
-If you are adding to a frontend that has a few tests already, skip the first
-two steps.
+#. Create a new file in the ``tests`` directory that will hold the new answer tests. The name of the file should begin with ``test_``.
 
-How to Write Image Comparison Tests
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#. Create a new class whose name begins with ``Test`` (e.g., ``TestTipsy``).
 
-We have a number of tests designed to compare images as part of yt. We make use
-of some functionality from matplotlib to automatically compare images and detect
-differences, if any. Image comparison tests are used in the plotting and volume
-rendering machinery.
+#. Decorate the class with ``pytest.mark.answer_test``. This decorator is used to tell pytest which tests are answer tests.
 
-The easiest way to use the image comparison tests is to make use of the
-``GenericImageTest`` class. This class takes three arguments:
+   .. note::
+
+      Tests that do not have this decorator are considered to be unit tests.
 
-* A dataset instance (e.g. something you load with ``yt.load`` or
-  ``data_dir_load``)
-* A function the test machinery can call which will save an image to disk. The
-  test class will then find any images that get created and compare them with the
-  stored "correct" answer.
-* An integer specifying the number of decimal places to use when comparing
-  images. A smaller number of decimal places will produce a less stringent test.
-  Matplotlib uses an L2 norm on the full image to do the comparison tests, so
-  this is not a pixel-by-pixel measure, and surprisingly large variations will
-  still pass the test if the strictness of the comparison is not high enough.
+#. Add the following three attributes to the class: ``answer_file=None``, ``saved_hashes=None``, and ``answer_version=000``. These attributes are used by the ``hashing`` fixture (discussed below) to automate the creation of new answer files as well as facilitate the comparison to existing answers.
 
-You *must* decorate your test function with ``requires_ds``, otherwise the
```

Comment:
is the `requires_ds` decorator no longer required? Looks like it's used in updated ImageComparison test example. If we still need it, would be nice to describe when it needs to be used.

Reply:
If the test parameterizes the ds (or list of datasets), then it isn't needed since the `ds` fixture will check to make sure the dataset exists. If it doesn't, the test is marked as failed. The decorator is still useful if you're not parameterizing the datasets, though. I'll make this clearer in the docs, thanks!

Reply:
wait. I thought we agreed to update the `requires_ds` decorator so that tests requiring missing data samples would be marked as SKIP (used to be PASS). Unless I'm missing an important difference between `requires_ds` and the new `ds` fixture, I think this is still the right thing to do (having tests shown as FAILLED while we don't even meet the requirements to run them feels like a very easy way to waste a lot of time spent in confusion).

coverage can later help us catch tests that are skipped on every run (across all OSes for example), and maybe _this_ should be considered a failure.

Reply:
Hey @jcoughlin11 I think this discussion still needs resolution

Reply:
> so I don't believe the requires_ds fixture is needed anymore.

I'm sure you mean the decorator. Well it's still used somewhere between 80 and 150 times throughout the code base so I don't get it. Is the plan to replace it with the `ds` fixture ?